src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -352,39 +358,70 @@ public ResponseEntity<ProgrammingExerciseStudentParticipation> resumeParticipati
 
-        if (programmingExercise.isExamExercise()) {
-            throw new BadRequestAlertException("Not intended for the use in exams", "participation", "preconditions not met");
+        Exercise exercise = exerciseRepository.findById(exerciseId).orElseThrow(() -> new ResourceNotFoundException("Exercise not found"));
+
+        if (!(exercise instanceof TextExercise) && !(exercise instanceof ProgrammingExercise)) {
+            throw new BadRequestAlertException("Unsupported exercise type", "participation", "unsupported type");
         }
 
-        if (programmingExercise.getDueDate() != null && now().isAfter(programmingExercise.getDueDate())) {
+        return handleExerciseFeedbackRequest(exercise, principal);
+    }
+
+    private ResponseEntity<StudentParticipation> handleExerciseFeedbackRequest(Exercise exercise, Principal principal) {
+        // Validate exercise and timing
+        if (exercise.isExamExercise()) {
+            throw new BadRequestAlertException("Not intended for the use in exams", "participation", "preconditions not met");
+        }
+        if (exercise.getDueDate() != null && now().isAfter(exercise.getDueDate())) {
             throw new BadRequestAlertException("The due date is over", "participation", "preconditions not met");
         }
+        if (exercise instanceof ProgrammingExercise) {
+            ((ProgrammingExercise) exercise).validateSettingsForFeedbackRequest();
+        }
 
-        var participation = programmingExerciseParticipationService.findStudentParticipationByExerciseAndStudentId(programmingExercise, principal.getName());
+        // Get and validate participation
         User user = userRepository.getUserWithGroupsAndAuthorities();
+        StudentParticipation participation = (exercise instanceof ProgrammingExercise)
+                ? programmingExerciseParticipationService.findStudentParticipationByExerciseAndStudentId(exercise, principal.getName())
+                : studentParticipationRepository.findByExerciseIdAndStudentLogin(exercise.getId(), principal.getName())
+                        .orElseThrow(() -> new ResourceNotFoundException("Participation not found"));
 
         checkAccessPermissionOwner(participation, user);
-        programmingExercise.validateSettingsForFeedbackRequest();
+        participation = studentParticipationRepository.findByIdWithResultsElseThrow(participation.getId());
 
-        var studentParticipation = (ProgrammingExerciseStudentParticipation) studentParticipationRepository.findByIdWithResultsElseThrow(participation.getId());
-        var result = studentParticipation.findLatestLegalResult();
-        if (result == null) {
-            throw new BadRequestAlertException("User has not reached the conditions to submit a feedback request", "participation", "preconditions not met");
+        // Check submission requirements
+        if (exercise instanceof TextExercise) {
```

Comment:
There are multiple differentiations between the different exercise types. I think it would be better to split the method into multiple parts, maybe even into different services (programmingExerciseService, ...)